### PR TITLE
fix for #1591

### PR DESCRIPTION
**Pull request by FrankMittelbach**
# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [x] Rollback provided (if necessary)?
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

**Review by davidcarlisle — APPROVED** [review]

File: base/doc/ltnews41.tex [hunk 1]
```diff
@@ -371,6 +371,18 @@ \subsection{Fix the use of  \texttt{localmathalpabets}}
 \githubissue[s]{1101 1028}
 
 
+\subsection{Prevent \texttt{cmd} hook from defining an undefined command}
+
+Using \verb=AddToHook{cmd/FOO/...}= when the command \cs{FOO} was
```
Comment (muzimuzhi) [inline base/doc/ltnews41.tex:376]:
```suggestion
Using \verb=\AddToHook{cmd/FOO/...}= when the command \cs{FOO} was
```

File: base/doc/ltnews41.tex [hunk 2]
```diff
@@ -371,6 +371,18 @@ \subsection{Fix the use of  \texttt{localmathalpabets}}
...
+undefined resulted in the command becoming \cs{relax}. Thus, if used,
+it no longer raised an \enquote{Undefined csname} error but silently
```
Comment (muzimuzhi) [inline base/doc/ltnews41.tex:378]:
```suggestion
it no longer raised an \enquote{Undefined control sequence} error but silently
```

File: base/update-lthooks-tests.sh [hunk 3]
```diff
@@ -34,7 +34,7 @@ l3build save -cconfig-lthooks2 \
 
 
 
-l3build save -cconfig-lthooks \
+L3build save -cconfig-lthooks \
```
Comment (muzimuzhi) [inline base/update-lthooks-tests.sh:37]:
```suggestion
l3build save -cconfig-lthooks \
```

**Review by muzimuzhi — APPROVED** [review]
Just typos.

File: base/update-lthooks-tests.sh (hunk 3 above)
Comment (FrankMittelbach) [reply base/update-lthooks-tests.sh:37]: ouch :-) comes from using ^C in emacs instead of the ^W

File: base/ltcmdhooks.dtx [hunk 4]
```diff
@@ -577,20 +577,53 @@
...
+%    nevertheless there to be consumed by \cs{@@_patch_cmd_or_delay:Nnn}.
+% \changes{v1.0k}{2024/12/22}{Avoid defining command while adding a cmd
+%    hook (gh/1591)}
+%    \begin{macrocode}
+    \group_begin:
+       \exp_after:wN
+    \group_end:
+       \exp_after:wN
+    \@@_patch_cmd_or_delay:Nnn
+       \cs:w #2\cs_end:
```
Comment (Skillmon) [inline base/ltcmdhooks.dtx:606]: Maybe use `\group_begin: \exp_args:NNc \group_end:` instead of the `\exp_after:wN` chain.

Comment (FrankMittelbach) [reply base/ltcmdhooks.dtx:606]: I'm not sure it is more readable this way (given that it is a hack anyway)

**Review by josephwright — APPROVED** [review]

**Review by car222222 — COMMENTED** [review]
No problems here by now. 

Seems I am too late to actually Approve it??

**Pull request merged**